srcalc/src/main/java/gov/va/med/srcalc/vista/vistalink/AccessVerifyConnectionSpec.java

```diff
@@ -0,0 +1,183 @@
+     * authentication. Not explicitly specified by VistALink, but used by
+     * KaajeeVistaLinkConnectionSpec. (See class Javadoc.)
+     */
+    public static final String SECURITY_TYPE = "av";
+    
+    /**
+     * The name of the Access/Verify code XML element.
+     */
+    private static final String ELEMENT_AV = "AccessVerify";
+    
+    private final String fAccessCode;
+    
+    private final String fClientIp;
+    
+    private final String fEncryptedComponents;
+    
+    private final HashCode fVerifyHash;
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            AccessVerifyConnectionSpec.class);
+    
+    /**
+     * Constructs an instance with the given properties.
+     * 
+     * @param division station # (external format) of the division to log the user in
+     * against
+     * @param accessCode user access code. Must not be null.
+     * @param verifyCode user verify code. Must not be null.
+     * @param clientIp the client's IP address for signon logging. Must not be null.
+     */
+    public AccessVerifyConnectionSpec(
+            final String division,
+            final String accessCode,
+            final String verifyCode,
+            final String clientIp)
+    {
+        super(division);
+        fAccessCode = Objects.requireNonNull(accessCode);
+        fClientIp = Objects.requireNonNull(clientIp);
+        Objects.requireNonNull(verifyCode);
+        final String components = accessCode + ";" + verifyCode + ";" + clientIp;
+        // Note that we do not store the plaint-text verify code to avoid careless
+        // observation in memory, but we do store a hash for equals() performance. Though
+        // MD5 would probably be fine here, use SHA-256 since it is superior.
+        fVerifyHash = Hashing.sha256().hashString(verifyCode, StandardCharsets.US_ASCII);
+
+        try
+        {
+            // Note: contrary to the class name, the returned string is not a hash or
+            // encrypted, but simply obfuscated by a reversible transformation.
+            fEncryptedComponents = VistaKernelHash.encrypt(components, true);
+        }
+        catch (VistaKernelHashCountLimitExceededException e)
+        {
+            // At time of writing, VistaKernelHashCountLimitExceededException should be
+            // so rare that its occurrence must be a data issue. So this exception is the
+            // best we can do.
+            throw new IllegalArgumentException("Could not encrypt access/verify code pair.");
+        }
+    }
+    
+    @Override
+    public ArrayList<String> getProprietarySecurityInfo()
+    {
+        // I am unsure what the significance of this returned list is. I am simply
+        // emulating the behavior of KaajeeVistaLinkConnectionSpec. - David Tombs, 25 Aug
+        // 2015
+        final ArrayList<String> values = new ArrayList<String>(1);
+        values.add(fEncryptedComponents);
+        return values;
```

Comment:
Definitely a bit interesting that you only need a single item here. Not sure what they were going for in the Kaajee class.